### PR TITLE
Hub - add hook for settings, feature flags & permissions

### PR DESCRIPTION
**Pull request by himdel**
Ensures all Hub pages have a context with settings, feature flags & permissions available.

(This makes it possible to do permission checking and feature flag checking in the hub portion of code. It doesn't attempt to add anything using it, and it does not attempt to bring this all the way to HubSidebar yet.)

---

Example usage:

```diff
--- a/frontend/hub/namespaces/HubNamespaceDetails.tsx
+++ b/frontend/hub/namespaces/HubNamespaceDetails.tsx
@@ -10,8 +10,11 @@ import { useHubNamespaceActions } from './hooks/useHubNamespaceActions';
 import { useHubNamespacesColumns } from './hooks/useHubNamespacesColumns';
 import { DropdownPosition } from '@patternfly/react-core';
 import { hubAPI } from '../api';
+import { useHubContext } from '../useHubContext';
+import { EmptyStateUnauthorized } from '../../../framework/components/EmptyStateUnauthorized';
 
 export function NamespaceDetails() {
+  const { hasPermission } = useHubContext();
   const { t } = useTranslation();
   const params = useParams<{ id: string }>();
   const { data } = useGet<HubItemsResponse<HubNamespace>>(
@@ -23,6 +26,11 @@ export function NamespaceDetails() {
   }
 
   const pageActions = useHubNamespaceActions();
+
+  if (!hasPermission('ansible.add_ansiblerepository')) {
+    return <EmptyStateUnauthorized />;
+  }
+
   return (
     <PageLayout>
       <PageHeader
```

Provides `user`, `featureFlags`, `settings`, `hasPermission`, and `errors`.
(leaving a TODO for later - expose errors in a toast if any)
(and possibly TODO more logic around errors and empty states)

### CODE REVIEW

**Review by ShaiahWren — APPROVED** [review]
This looks good so far, thanks for adding :)

File: frontend/hub/useHubContext.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        getFeatureFlags.error,
+        getSettings.error,
+        getUser.error,
+        ...(getFeatureFlags.data?._messages || []),
+      ].filter(Boolean) as string[],
+      featureFlags: getFeatureFlags.data as HubFeatureFlags,
+      settings: getSettings.data as HubSettings,
+      user: getUser.data as HubUser,
+    } as HubContext);
+  }, [getFeatureFlags, getSettings, getUser]);
```
Comment (vidyanambiar) [inline frontend/hub/useHubContext.tsx:110]: I tried testing by adding the line `const hubConfig = useHubContext();` in the Collections component and it was causing the page to get blocked due to too many re-renders.
image

```suggestion
  }, [
    getFeatureFlags.isLoading,
    getSettings.isLoading,
    getUser.isLoading,
    getFeatureFlags.data,
    getSettings.data,
    getUser.data,
  ]);
```

**Review by vidyanambiar — CHANGES_REQUESTED** [review]
I added a comment related to the dependency array for the HubContextProvider's useEffect hook. Other than that this looks good to me 👍

File: frontend/hub/useHubContext.tsx (hunk 1 above)
Comment (himdel) [reply frontend/hub/useHubContext.tsx:110]: ah, I had it that way with `isLoading` but then the linter forced me to remove it or include all the .data and .error fields as well :)

Sounds like I have to do the latter, thanks :)

**Review by MilanPospisil — APPROVED** [review]
LGTM :+1:

**Comment by himdel** [comment]
@vidyanambiar looks like I can't merge without you ..doing something to the change request?

**Review by vidyanambiar — APPROVED** [review]
lgtm 👍

**Pull request merged**